mesh/mesh_test.go

```diff
@@ -383,18 +379,15 @@ func TestMesh_ExtractUniqueOrderedTransactions(t *testing.T) {
 	tx2 := addTxToMesh(r, msh, signer, 2469)
 	tx3 := addTxToMesh(r, msh, signer, 2470)
 	tx4 := addTxToMesh(r, msh, signer, 2471)
-	tx5 := addTxToMesh(r, msh, signer, 2472)
 	addBlockWithTxs(r, msh, layerID, true, tx1, tx2)
 	addBlockWithTxs(r, msh, layerID, true, tx2, tx3, tx4)
-	addBlockWithTxs(r, msh, layerID, false, tx4, tx5)
-	addBlockWithTxs(r, msh, layerID, false, tx5)
+	addBlockWithTxs(r, msh, layerID, false, tx4)
 	l, err := msh.GetLayer(layerID)
 	r.NoError(err)
 
-	validBlocks, invalidBlocks := msh.ExtractUniqueOrderedTransactions(l)
+	validBlocks := msh.ExtractUniqueOrderedTransactions(l)
 
 	r.ElementsMatch(GetTransactionIds(tx1, tx2, tx3, tx4), GetTransactionIds(validBlocks...))
-	r.ElementsMatch(GetTransactionIds(tx5), GetTransactionIds(invalidBlocks...))
```

Comment:
This is some pretty delicate logic that was seemingly lost :(

Reply:
Let's recreate this test with the new logic (essentially test that `reInsertTxsToPool` does the right thing with all the edge cases that were originally in this test (all kinds of overlapping transactions between valid/invalid blocks).

Reply:
I still don't see that this is tested. I can live with leaving it untested if you feel this would be hard with the new way this is written, because I expect this logic to change in the next couple of months, but I would sleep better if this logic was tested, so just making sure this wasn't just forgotten.

By "this logic" I mean: if a transaction appears in both a valid and an invalid block (`tx4`) - it's treated only as valid, but if a transaction appears only in an invalid block (`tx5`) it's treated only as invalid. Txs that appear in only one (`tx1`, `tx3`) or more (`tx2`) valid blocks are also treated as valid, and only handled once.